src/OSPSuite.Core/Domain/Services/SimModelManager.cs

```diff
@@ -48,19 +42,25 @@ private void raiseSimulationProgress(int progress)
 
       public SimulationRunResults RunSimulation(IModelCoreSimulation simulation, SimulationRunOptions simulationRunOptions = null)
       {
+         return RunSimulationAsync(simulation, simulationRunOptions: simulationRunOptions).GetAwaiter().GetResult();
+      }
+
+      public async Task<SimulationRunResults> RunSimulationAsync(IModelCoreSimulation simulation, CancellationToken cancellationToken = default, SimulationRunOptions simulationRunOptions = null)
+      {
+         _simulationRunOptions = simulationRunOptions ?? new SimulationRunOptions();
          try
          {
-            _simulationRunOptions = simulationRunOptions ?? new SimulationRunOptions();
-            _canceled = false;
-            doIfNotCanceled(() => loadSimulation(simulation));
-            doIfNotCanceled(() => FinalizeSimulation(_simModelSimulation));
-            doIfNotCanceled(simulate);
+            await loadAndRunSimulationAsync(simulation, cancellationToken);
 
-            if(!_canceled)
-               return new SimulationRunResults(WarningsFrom( _simModelSimulation), getResults(simulation));
+            if (!cancellationToken.IsCancellationRequested)
```

Comment:
I think this is not required. If you arrive here, we have results anyways so we should return it. If it was canceled, we are in the catch bloxk. So the if is not required and line 58 either